src/plugins/intel_cpu/src/emitters/snippets/riscv64/jit_loop_emitters.cpp

```diff
@@ -0,0 +1,307 @@
+    work_amount = loop_end->get_work_amount();
+    wa_increment = loop_end->get_increment();
+    evaluate_once = loop_end->get_evaluate_once();
+    loop_id = loop_end->get_id();
+    is_work_amount_dynamic = ov::snippets::utils::is_dynamic_value(work_amount);
+    OPENVINO_ASSERT(wa_increment > 0, "Loop increment must be > 0");
+
+    loop_begin_label = std::make_shared<Xbyak_riscv::Label>();
+    loop_end_label = nullptr;
+
+    // LoopBegin communicates work_amount via GPR to LoopEnd
+    in_out_type_ = emitter_in_out_map::gpr_to_gpr;
+}
+
+void jit_loop_begin_emitter::validate_arguments(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
+    OV_CPU_JIT_EMITTER_ASSERT(in.empty(), "Invalid inputs size: expected 0 got " + std::to_string(in.size()));
+    // The only expected output is work amount register (communicated to jit_loop_end_emitter)
+    OV_CPU_JIT_EMITTER_ASSERT(out.size() == 1, "Invalid outputs size: expected 1 got " + std::to_string(out.size()));
+}
+
+void jit_loop_begin_emitter::emit_code_impl(const std::vector<size_t>& in,
+                                            const std::vector<size_t>& out,
+                                            const std::vector<size_t>& pool_vec_idxs,
+                                            const std::vector<size_t>& pool_gpr_idxs) const {
+    validate_arguments(in, out);
+    // Use base preamble/postamble to manage aux regs consistently
+    ov::intel_cpu::riscv64::jit_emitter::emit_code_impl(in, out, pool_vec_idxs, pool_gpr_idxs, {});
+}
+
+void jit_loop_begin_emitter::emit_impl([[maybe_unused]] const std::vector<size_t>& in,
+                                       const std::vector<size_t>& out) const {
+    auto reg_work_amount = Xbyak_riscv::Reg(out[0]);
+    if (is_work_amount_dynamic) {
+        const auto id_offset = loop_id * sizeof(ov::intel_cpu::jit_snippets_call_args::loop_args_t);
+        // Acquire two scratch regs
+        std::vector<size_t> used = {out[0]};
+        jit_aux_gpr_holder h_ptr(h, aux_gpr_idxs, used);
+        jit_aux_gpr_holder h_tmp(h, aux_gpr_idxs, used);
+        auto reg_loop_args_ptr = h_ptr.get_reg();
+        auto addr = h_tmp.get_reg();
+        // reg_loop_args_ptr = *(a0 + GET_OFF(loop_args))
+        h->uni_li(addr, GET_OFF(loop_args));
+        h->add(addr, Xbyak_riscv::a0, addr);
+        h->ld(reg_loop_args_ptr, addr, 0);
+        // reg_loop_args_ptr += id_offset + OFF(m_work_amount)
+        h->uni_li(addr, id_offset + GET_OFF_LOOP_ARGS(m_work_amount));
+        h->add(reg_loop_args_ptr, reg_loop_args_ptr, addr);
+        // load m_work_amount
+        h->ld(reg_work_amount, reg_loop_args_ptr, 0);
+    } else {
+        h->uni_li(reg_work_amount, static_cast<size_t>(work_amount));
+    }
+    // If evaluate_once and not dynamic increment, skip branch to end and simply mark begin
+    if (evaluate_once && !ov::snippets::utils::is_dynamic_value(wa_increment)) {
+        if (!begin_label_bound) {
```

Comment:
Why do we need `begin_label_bound`? It is always false before we call this method, it always become true at L144, and then the if on L157 is always false. Did I miss something?